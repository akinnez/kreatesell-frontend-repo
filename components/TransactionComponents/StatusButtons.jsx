import {useEffect} from 'react';

import {Button} from 'antd';
import styles from './header.module.scss';

const statusArr = [
	{type: 'All', label: 'All'},
	{type: 'Successful', label: 'Successful'},
	{type: 'Cleared', label: 'Cleared'},
	{type: 'Cancelled', label: 'Cancelled'},
	{type: 'Failed', label: 'Failed'},
];

const StatusButtons = ({setFilters, filters, setLoading}) => {
	useEffect(() => {
		setFilters((prev) => ({...prev, status: 'Successful'}));
	}, []);

	const handleClick = (type) => {
		setFilters({...filters, status: type});
		setLoading(true);
	};

	return (
		<section className={styles.status__btns__section}>
			<div className={styles.status__btns}>
				{statusArr.map(({type, label}) => (
					<div className={styles.status__btn} key={label}>
						<Button
							onClick={() => handleClick(type)}
							type={
								filters.status === type ? 'primary' : 'default'
							}
						>
							{label}
						</Button>
					</div>
				))}
			</div>
		</section>
	);
};

export default StatusButtons;
