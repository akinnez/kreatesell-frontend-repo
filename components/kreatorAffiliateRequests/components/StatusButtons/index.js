import { Button } from "antd";
import styles from "./index.module.scss";

const statusArr = [
  { type: "All", label: "All" },
  { type: "Pending", label: "Pending" },
  { type: "Approved", label: "Approved" },
  { type: "Declined", label: "Declined" },
];

const StatusButtons = ({ setFilters, filters }) => {
  const handleClick = type => {
    setFilters({ ...filters, status: type });
  };

  return (
    <section className={styles.status__btns__section}>
      <div className={styles.status__btns}>
        {statusArr.map(({ type, label }) => (
          <div className={styles.status__btn} key={label}>
            <Button
              onClick={() => handleClick(type)}
              type={filters.status === type ? "primary" : "default"}
            >
              {label}
            </Button>
          </div>
        ))}
      </div>
    </section>
  );
};

export default StatusButtons;
