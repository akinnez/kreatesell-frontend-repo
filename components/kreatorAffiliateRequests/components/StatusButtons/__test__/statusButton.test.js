import StatusButtons from "..";
import renderer from "react-test-renderer";

const mockProps = {
  setFilters: jest.fn(),
  filters: {},
  setLoadng: jest.fn(),
};

it("renders a snapshot of <StatusButtons />", () => {
  const button = renderer.create(<StatusButtons {...mockProps} />).toJSON();
  expect(button).toMatchSnapshot();
});
