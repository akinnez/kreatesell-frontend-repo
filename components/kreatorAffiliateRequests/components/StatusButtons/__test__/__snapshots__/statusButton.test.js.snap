// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <StatusButtons /> 1`] = `
<section
  className="status__btns__section"
>
  <div
    className="status__btns"
  >
    <div
      className="status__btn"
    >
      <button
        className="ant-btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          All
        </span>
      </button>
    </div>
    <div
      className="status__btn"
    >
      <button
        className="ant-btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          Pending
        </span>
      </button>
    </div>
    <div
      className="status__btn"
    >
      <button
        className="ant-btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          Approved
        </span>
      </button>
    </div>
    <div
      className="status__btn"
    >
      <button
        className="ant-btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          Declined
        </span>
      </button>
    </div>
  </div>
</section>
`;
